dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/properties/ModulePropertiesUtils.java

```diff
@@ -227,15 +228,15 @@ protected ComponentMetaData getComponentMetaData() {
 
     @SuppressWarnings("unchecked")
     private Bean<HttpAuthenticationMechanism> getBean(BeanManager beanManager, Class<HttpAuthenticationMechanism> hamClass) {
+        Bean<HttpAuthenticationMechanism> bean = null;
         Set<Bean<?>> beans = beanManager.getBeans(hamClass);
         if (beans.size() == 1) {
-            return (Bean<HttpAuthenticationMechanism>) beans.iterator().next();
-        } else {
-            if (tc.isDebugEnabled()) {
-                Tr.debug(tc, "Number of HAMs : " + beans.size());
-            }
+            bean = (Bean<HttpAuthenticationMechanism>) beans.iterator().next();
         }
-        return null;
+        if (tc.isDebugEnabled()) {
+            Tr.debug(tc, "HAM : " + hamClass + " Number of HAM bean: " + beans.size() + " bean.toString(): " + (bean != null ? beans.iterator().next().toString() : "<NO bean>"));
```

Comment:
This can cause an NPE
```suggestion
            Tr.debug(tc, "HAM : " + hamClass + " Number of HAM bean: " + beans.size() + " bean.toString(): " + (bean != null ? bean.toString() : "<NO bean>"));
```

Reply:
I agree with Zech's suggestion.